components/root/root.jsx

```diff
@@ -147,7 +149,75 @@ export default class Root extends React.Component {
     }
 
     handleAccessibilityKeys = (e) => {
-        if (isKeyPressed(e, Constants.KeyCodes.TAB)) {
+        if (isKeyPressed(e, Constants.KeyCodes.DOWN)) {
+            const lhsList = document.getElementById('lhsList');
+            if (lhsList.contains(e.target)) {
+                this.sidebarCategories = [
+                    document.getElementById('unreadsChannelList'),
+                    document.getElementById('favoriteChannelList'),
+                    document.getElementById('publicChannelList'),
+                    document.getElementById('privateChannelList'),
+                    document.getElementById('directChannelList'),
+                ].filter((element) => Boolean(element));
+
+                const containsElement = (element, i, elementsList) => {
+                    if (element.contains(e.target)) {
+                        this.currentSidebarFocus = i + 1;
+                        return true;
+                    } else if (element.contains(e.target) && i === elementsList.length - 1) {
+                        this.currentSidebarFocus++;
+                        return false;
+                    }
+                    return false;
+                };
+
+                if (!this.sidebarCategories.some(containsElement)) {
+                    this.currentSidebarFocus = 0;
+                }
+
+                if (this.currentSidebarFocus === this.sidebarCategories.length) {
+                    this.currentSidebarFocus = 0;
+                }
+
+                const activeElement = this.sidebarCategories[this.currentSidebarFocus];
+                this.addFocusClass(activeElement);
+                this.currentSidebarFocus++;
+            }
+        } else if (isKeyPressed(e, Constants.KeyCodes.UP)) {
+            const lhsList = document.getElementById('lhsList');
+            if (lhsList.contains(e.target)) {
+                this.sidebarCategories = [
+                    document.getElementById('unreadsChannelList'),
+                    document.getElementById('favoriteChannelList'),
+                    document.getElementById('publicChannelList'),
+                    document.getElementById('privateChannelList'),
+                    document.getElementById('directChannelList'),
+                ].filter((element) => Boolean(element));
```

Comment:
A lot of this and the below code is duplicated. Can you factor out the common code to share or re-use it?